src/db.rs

```diff
@@ -96,4 +109,28 @@ impl Database {
     pub fn insert_witnesses(&self) -> Result<()> {
         unimplemented!();
     }
+
+    pub fn create_database_if_necessary(&self) -> Result<()> {
+        let path_buf: PathBuf = get_app_root(AppDataType::UserData, &APP_INFO)?;
+
+        let mut initial_db_path_buf: PathBuf = path_buf.clone();
+        initial_db_path_buf.push(INITIAL_DB_NAME);
+
+        let initial_db_path: &Path = initial_db_path_buf.as_path();
+
+        let mut db_path_buf: PathBuf = path_buf.clone();
+        db_path_buf.push(DB_NAME);
+        let db_path: &Path = db_path_buf.as_path();
+
+        let cur_path: &Path = path_buf.as_path();
+        let parent_dir: &Path = cur_path.parent().unwrap();
+        if db_path.exists() == false {
+            fs::create_dir(parent_dir)?;
```

Comment:
why do we need to create parent_dir and cur_path, aren't them already there? if not can we create them in a single path by using `create_dir_all`